animating-widgets.md

```diff
@@ -0,0 +1,293 @@
+available.
+
+The [`flutter/animation`](https://docs.flutter.io/flutter/animation/animation-library.html)
+package in the Flutter SDK provides APIs to create animation effects in user
+interfaces (UIs) that are built on the
+[`StatelessWidget`](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html)
+and [`StatefulWidget`](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html)
+classes.
+
+Animations in Flutter are encapsulated as
+[`Animation`](https://docs.flutter.io/flutter/animation/Animation-class.html)
+objects that contain these key properties:
+
++ **Value:** The animation object’s typed value represents a data point within
+a range (also called an interpolation). You can define an animation class to
+generate values based on a given functional mapping, such as a linear, curve,
+or step-wise function.
++ **Status:** The animation object’s current status (such as forward,
+backward, completed, and dismissed), as listed by the
+[`AnimationStatus`](https://docs.flutter.io/flutter/animation/AnimationStatus-class.html)
+constants.
+
+To create an animation effect in your app, you need to write code to do the
+following:
+
+1. Attach an animation object to a widget or a listener to detect changes to the
+animation object.
+2. Specify how the the animation is *interpolated* (that is, how the animation
+objects’s value and status changes over
+time) as it runs.
+
+Based on the animation object’s properties at a specific frame, Flutter
+dynamically modifies your widget visual appearence, and rebuilds the widget
+tree to produce visual transitions such as sliding, scaling, rotation, height
+or width changes, and fading.
+
+Flutter's flexible animation model allows you to define interpolations for
+spatial positioning, color gradients, image scaling, and others. You can define
+your own subclass of
+[`Animation`](https://docs.flutter.io/flutter/animation/Animation-class.html)
+or use an animation subclass that the framework provides.
+
+To learn more about
+using the other key classes in the
+[`flutter/animation`](https://docs.flutter.io/flutter/animation/animation-library.html)
+package, see the [Animations in Flutter](/animations) guide.
+
+## Render a basic widget animation
+
+To render your widget with an animation, set one or more of the widget’s
+members to the animation object, then use the animation’s value to decide how
+to draw your object.
+
+If you are animating a stateful widget, you can notify your widget of changes
+to the animation’s values by using the
+[`addListener()`](https://docs.flutter.io/flutter/animation/Animation/addListener.html)
+method. The framework invokes the listener callback whenever the value of the
+animation changes. Typically, you’ll want to call the
+[`setState()`](http://docs.flutter.io/flutter/material/State/setState.html)
+method when the animation value change occurs, to trigger the framework to
+rebuild the widget tree.
```

Comment:
Rather than doing this, we might want to recommend that people use `AnimatedWidget` or `AnimatedBuilder`.